oap-ape/ape-native/src/utils/AggExpression.h

```diff
@@ -416,14 +418,26 @@ class Add : public ArithmeticExpression {
     result.precision = precision;
     result.scale = scale;
     if (left.data.size() == 1) {
+      arrow::Decimal128 leftValue = left.data[0];
+
+      // rescale may be needed. e.g. literal value in SQL
+      if (left.scale != result.scale) {
+        leftValue = leftValue.Rescale(left.scale, result.scale).ValueOrDie();
+      }
       for (int i = 0; i < right.data.size(); i++) {
-        arrow::BasicDecimal128 out = left.data[0] + right.data[i];
+        arrow::BasicDecimal128 out = leftValue + right.data[i];
         result.data.push_back(out);
       }
       result.nullVector = right.nullVector;
     } else if (right.data.size() == 1) {
+      arrow::Decimal128 rightValue = right.data[0];
+
+      // rescale may be needed. e.g. literal value in SQL
+      if (right.scale != result.scale) {
+        rightValue = rightValue.Rescale(right.scale, result.scale).ValueOrDie();
```

Comment:
should we consider two columns' scale/precision not equal case?

Reply:
Does `Multiply` node need this fix?

Reply:
According to my observation in Flink, only scale matters here, and the result scale will be the same as one of the input column.

`Multiply` does not have scale problem in Flink's agg expressions.